ingestion/src/metadata/utils/datalake_utils.py

```diff
@@ -0,0 +1,376 @@
+                ),
+                dataframe=DataFrame.from_records(elements),
+            )
+        return DatalakeColumnWrapper(dataframes=DataFrame.from_records(elements))
+    except (AssertionError, InvalidAvroBinaryEncoding):
+        columns = parse_avro_schema(schema=avro_text, cls=Column)
+        field_map = {
+            col.name.__root__: Series(PD_AVRO_FIELD_MAP.get(col.dataType.value, "str"))
+            for col in columns
+        }
+        return DatalakeColumnWrapper(columns=columns, dataframes=DataFrame(field_map))
+
+
+def return_azure_storage_options(config_source: Any) -> Dict:
+    connection_args = config_source.securityConfig
+    return {
+        "tenant_id": connection_args.tenantId,
+        "client_id": connection_args.clientId,
+        "account_name": connection_args.accountName,
+        "client_secret": connection_args.clientSecret.get_secret_value(),
+    }
+
+
+def read_from_pandas(path: str, separator: str, storage_options=None):
+    chunk_list = []
+    with pd.read_csv(
+        path, sep=separator, chunksize=CHUNKSIZE, storage_options=storage_options
+    ) as reader:
+        for chunks in reader:
+            chunk_list.append(chunks)
+    return chunk_list
+
+
+def dataframe_to_chunks(df):
+    """
+    Reads the Dataframe and returns list of dataframes broken down in chunks
+    """
+    return [
+        df[range_iter : range_iter + CHUNKSIZE]
+        for range_iter in range(0, len(df), CHUNKSIZE)
+    ]
+
+
+def _get_json_text(key: str, text: bytes, decode: bool) -> str:
+    if key.endswith(".gz"):
+        return gzip.decompress(text)
+    if key.endswith(".zip"):
+        with zipfile.ZipFile(io.BytesIO(text)) as zip_file:
+            return zip_file.read(zip_file.infolist()[0]).decode(UTF_8)
+    if decode:
+        return text.decode(UTF_8)
+    return text
+
+
+def read_from_json(key: str, json_text: str, decode: bool = False) -> List:
+    """
+    Read the json file from the azure container and return a dataframe
+    """
+
+    # pylint: disable=import-outside-toplevel
+    from pandas import json_normalize
+
+    json_text = _get_json_text(key, json_text, decode)
+    try:
+        data = json.loads(json_text)
+    except json.decoder.JSONDecodeError:
+        logger.debug("Failed to read as JSON object trying to read as JSON Lines")
+        data = [json.loads(json_obj) for json_obj in json_text.strip().split("\n")]
+    return dataframe_to_chunks(json_normalize(data, sep=COMPLEX_COLUMN_SEPARATOR))
+
+
+@singledispatch
+def read_csv_dispatch(config_source: Any, **kwargs):
+    raise NotImplementedError(
+        f"Didn't Implement {config_source.__class__.__name__} for CSV"
+    )
+
+
+@singledispatch
+def read_tsv_dispatch(config_source: Any, **kwargs):
+    raise NotImplementedError(
+        f"Didn't Implement {config_source.__class__.__name__} for TSV"
+    )
+
+
+@singledispatch
+def read_avro_dispatch(config_source: Any, **kwargs):
+    raise NotImplementedError(
```

Comment:
Instead of calling the same exception and just changing the final name, let's create a Custom Exception that raises this by eating the format as an input param.

Also I think the `Didn't Implement...` reads a bit weird. I'd do something like `Missing implementation for source X and format Y`